versioned/dynamodb/src/main/java/com/dremio/nessie/versioned/implrocksdb/RocksDBStoreManager.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.implrocksdb;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.DBOptions;
+import org.rocksdb.Options;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.dremio.nessie.versioned.ReferenceNotFoundException;
+import com.dremio.nessie.versioned.impl.InternalRef;
+import com.dremio.nessie.versioned.impl.condition.ConditionExpression;
+import com.dremio.nessie.versioned.impl.condition.UpdateExpression;
+import com.dremio.nessie.versioned.store.Id;
+import com.dremio.nessie.versioned.store.LoadStep;
+import com.dremio.nessie.versioned.store.SaveOp;
+import com.dremio.nessie.versioned.store.Store;
+import com.dremio.nessie.versioned.store.ValueType;
+import com.google.common.base.Function;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.cache.RemovalListener;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+public class RocksDBStoreManager implements Store {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RocksDBStoreManager.class);
+
+  public static final byte[] REF_COLUMN_FAMILY = "ref".getBytes();
+  public static final byte[] L1_COLUMN_FAMILY = "l1".getBytes();
+  public static final byte[] L2_COLUMN_FAMILY = "l2".getBytes();
+  public static final byte[] L3_COLUMN_FAMILY = "l3".getBytes();
+  public static final byte[] VALUE_COLUMN_FAMILY = "value".getBytes();
+  public static final byte[] KEY_FRAGMENT_COLUMN_FAMILY = "keyfragment".getBytes();
+  public static final byte[] COMMIT_METADATA_AUTHOR_COLUMN_FAMILY = "commitmetadataauthor".getBytes();
+  public static final byte[] COMMIT_METADATA_MESSAGE_COLUMN_FAMILY = "commitmetadatamessage".getBytes();
+  private static ArrayList<byte[]> COLUMN_FAMILIES = new ArrayList<byte[]>() {{
```

Comment:
Should store List<> instead of ArrayList - always have the variables be the most generic level of interface possible.

Reply:
Agreed. Good catch, I'll update.